providers/azure.go

```diff
@@ -128,3 +133,138 @@ func (p *AzureProvider) GetEmailAddress(s *sessions.SessionState) (string, error
 
 	return email, err
 }
+
+// Redeem exchanges the OAuth2 authentication token for an ID token
+func (p *AzureProvider) Redeem(redirectURL, code string) (s *sessions.SessionState, err error) {
```

Comment:
Simply it is not (actually I used google provider, but changed to oidc provider in recent commit). All the providers are using different ways of doing similar things, so it could be unified for sure. I also just wanted to make it work as easy as possible. I can definitely work on it, but not sure if this should be part of this PR.

Reply:
@JoelSpeed would you like me to unify/refactor providers in this PR?

Reply:
I think unifying/refactoring should possibly be left for a separate PR (to keep the changeset small). If you're interested on working on that I'd definitely appreciate the PR though, if not, I think we should open an issue and hopefully someone will do it at a later date